index.js

```diff
@@ -173,7 +173,14 @@ function compression(options) {
 
       // compression method
       var accept = accepts(req)
-      var method = accept.encoding(['gzip', 'deflate', 'identity'])
+      var method = null;
+
+      // If we support a custom encoding and a custom encoding was requested
+      if (opts.compressor) {
+        method = accept.encoding(Object.keys(opts.compressor));
```

Comment:
Unfortunately it looks like there is still a bug here. The header `Accept-Encoding: br;q=0.001, gzip, deflate` will incorrectly use `br` over `gzip` just because the user wanted to provide an implementation for `br`.